nft_ingester/src/processors/transaction_based/bubblegum_updates_processor.rs

```diff
@@ -274,134 +281,126 @@ impl BubblegumTxProcessor {
-            | InstructionName::Delegate => {
+            | InstructionName::Delegate
+            | InstructionName::DelegateV2
+            | InstructionName::DelegateAndFreezeV2
+            | InstructionName::FreezeV2
+            | InstructionName::SetNonTransferableV2
+            | InstructionName::ThawV2
+            | InstructionName::ThawAndRevokeV2 => {
                 Self::get_update_owner_update(parsing_result, bundle).map(From::from).map(Ok)?
             },
-            InstructionName::Burn => {
+            InstructionName::Burn | InstructionName::BurnV2 => {
                 Self::get_burn_update(parsing_result, bundle).map(From::from).map(Ok)?
             },
-            InstructionName::MintV1 | InstructionName::MintToCollectionV1 => {
-                Self::get_mint_v1_update(parsing_result, bundle.slot).map(From::from).map(Ok)?
+            InstructionName::MintV1
+            | InstructionName::MintToCollectionV1
+            | InstructionName::MintV2 => {
+                Self::get_mint_v1_v2_update(parsing_result, bundle.slot).map(From::from).map(Ok)?
             },
             InstructionName::Redeem => {
                 Self::get_redeem_update(parsing_result, bundle).map(From::from).map(Ok)?
             },
             InstructionName::DecompressV1 => Ok(Self::get_decompress_update(bundle).into()), // no change log here? really?
-            InstructionName::VerifyCreator | InstructionName::UnverifyCreator => {
+            InstructionName::VerifyCreator
+            | InstructionName::UnverifyCreator
+            | InstructionName::VerifyCreatorV2
+            | InstructionName::UnverifyCreatorV2 => {
                 Self::get_creator_verification_update(parsing_result, bundle)
                     .map(From::from)
                     .map(Ok)?
             },
             InstructionName::VerifyCollection
             | InstructionName::UnverifyCollection
-            | InstructionName::SetAndVerifyCollection => {
+            | InstructionName::SetAndVerifyCollection
+            | InstructionName::SetCollectionV2 => {
                 Self::get_collection_verification_update(parsing_result, bundle)
                     .map(From::from)
                     .map(Ok)?
             },
-            InstructionName::UpdateMetadata => {
+            InstructionName::UpdateMetadata | InstructionName::UpdateMetadataV2 => {
                 Self::get_update_metadata_update(parsing_result, bundle).map(From::from).map(Ok)?
             },
-            InstructionName::FinalizeTreeWithRoot
-            | InstructionName::FinalizeTreeWithRootAndCollection => {
-                Self::get_create_tree_with_root_update(parsing_result, bundle)
-                    .map(From::from)
-                    .map(Ok)?
+            InstructionName::UpdateAssetDataV2 => {
+                debug!("Bubblegum: UpdateAssetDataV2 Not Implemented Instruction");
+                Ok(InstructionResult::default())
```

Comment:
It's not ok having the non implemented instruction here. It'll update the tree leaf and the proof, the data_hash will be updated. So this should be handled for sure, do we plan to do that as a separate effort? The contract already has the implementation.

Reply:
I took this from the old [das implementation](https://github.com/metaplex-foundation/digital-asset-rpc-infrastructure/pull/252/files#diff-2c0b4a5e64f3aafa5d5dfcd67dcd0b7d6adbb1c6d0a7900f33a8e454de617ae6R133). 
I think it will be implemented later